lib/_sendEvent.ts

```diff
@@ -1,66 +1,68 @@
-import { ClickReport } from './click';
-import { ConversionReport } from './conversion';
-import { SearchReport } from './search';
+import { InsightsSearchClickEvent } from './click';
+import { InsightsSearchConversionEvent } from './conversion';
 
-declare var process : {
+declare var process: {
   env: {
-    NODE_ENV: string
-  }
-}
+    NODE_ENV: string;
+  };
+};
 
-export type AnalyticsEvent = {
+export type InsightsEventTypes = 'click' | 'conversion';
+export type InsightsEvent = {
+  eventType: InsightsEventTypes;
+  eventName: string;
   userID: string;
   timestamp: number;
   indexName: string;
-}
-
-export type ReportEvent = 'search'|'click'|'conversion'
+};
 
 /**
  *  Sends data to endpoint
- * @param eventType 'click'|'conversion'
- * @param eventData ConversionReport|ClickReport
+ * @param eventType InsightsEventTypes
```

Comment:
nit: not sure where you defined it, but this should be singular: `type InsightsEventType = 'click' | 'conversion' ...`